crossfit/backend/cudf/series.py

```diff
@@ -12,9 +12,72 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+from functools import lru_cache
+from typing import TYPE_CHECKING, Any, Optional
+
 import cudf
 import cupy as cp
 from cudf.core.column import as_column
+from cudf.core.dtypes import ListDtype
+
+if TYPE_CHECKING:
+    from cudf.core.buffer import Buffer
+    from cudf.core.column import ColumnBase
+    from cudf.core.column.numerical import NumericalColumn
+
+
+@lru_cache
```

Comment:
I added tests for the following method

```python
@pytest.mark.parametrize("mock_version_value, expected", [
    ("24.10.0a1", True),   # Pre-release version of 24.10
    ("24.10.0", True),     # Version exactly 24.10
    ("24.10.3", True),     # Minor Version above 24.10
    ("25.0.0", True),      # Version above 24.10
    ("24.8.0", False),     # Version below 24.10
    ("24.8.0a1", False),   # Pre-release version below 24.10
])
def test_is_cudf_gte_24_10(mock_version_value, expected):
    with patch.object(cudf, '__version__', mock_version_value) as _:
        assert _is_cudf_gte_24_10() == expected
```


However it runs into interesting errors the mock influences the behavior for the remaining tests depending on the order / which tests ran.